src/pool/builder.rs

```diff
@@ -178,36 +178,21 @@ impl Builder {
     /// Freezes the configurations and returns the task scheduler and
     /// a builder to for lazy spawning threads.
     ///
-    /// It internally uses single level queue to setup the pool.
+    /// If `None` is supplied, it uses single level queue to setup the pool.
     ///
     /// In some cases, especially building up a large application, a task
     /// scheduler is required before spawning new threads. You can use this
     /// to separate the construction and starting.
-    pub fn freeze<T>(&self) -> (Remote<T>, LazyBuilder<T>)
-    where
-        T: TaskCell + Send,
-    {
-        self.freeze_with_queue(queue::single_level)
-    }
-
-    /// Freezes the configurations and returns the task scheduler and
-    /// a builder to for lazy spawning threads.
-    ///
-    /// `queue_builder` is a closure that creates a task queue. It accepts the
-    /// number of local queues and returns the task injector and local queues.
-    ///
-    /// In some cases, especially building up a large application, a task
-    /// scheduler is required before spawning new threads. You can use this
-    /// to separate the construction and starting.
-    pub fn freeze_with_queue<T>(
-        &self,
-        queue_builder: impl FnOnce(usize) -> (TaskInjector<T>, Vec<LocalQueue<T>>),
-    ) -> (Remote<T>, LazyBuilder<T>)
+    pub fn freeze<T>(&self, multilevel: Option<multilevel::Builder>) -> (Remote<T>, LazyBuilder<T>)
```

Comment:
I don't think it's a good idea to put a `multilevel` parameter here. It is not extensible at all. 

If we really don't want to expose `TaskInjector` and `LocalQueue`, can we give this builder a trait while still not exposing the queue types? (Rustdoc will show them as black and the user has no ability to implement the trait)

Reply:
How about using a unify builder?

Reply:
Then, in order not to expose the queue type, we will have a builder with no `build` function visible to users (this pr is the same)...It seems a bit strange to me but acceptable.